### ISSUE TITLE
Collections back-end: Tech spec review

### ISSUE THREAD

**Issue by richaagarwal** [chanzuckerberg/napari-hub#618]
Review tech spec - https://docs.google.com/document/d/1-ATD0ykdvSa1NTUL8XeE4FiHFe4jDGcFapzaNnQgi6s/edit# - and come up with implementation plan. 

Deliverable of this spike should be to fill in more implementation details into this ticket https://github.com/chanzuckerberg/napari-hub/issues/506 or break it out into a few smaller implementation tickets.

**Comment by richaagarwal** [comment]
@richaagarwal and @klai95 will have a working session on the tech spec

**Comment by richaagarwal** [comment]
I have updated the tech spec with an additional section, "Alternative to S3 cache", viewable here - https://docs.google.com/document/d/1-ATD0ykdvSa1NTUL8XeE4FiHFe4jDGcFapzaNnQgi6s/edit#heading=h.day8153r64bf

I have updated two tickets, https://github.com/chanzuckerberg/napari-hub/issues/506 and https://github.com/chanzuckerberg/napari-hub/issues/507, with the relevant information. 

I still need to create an additional ticket to reflect a caching approach (via possibly CloudFront), and then we should have the tech spec re-reviewed by additional colleagues. I'll leave this open for now, but the above two tickets are considered "Ready" as they will help inform our proof of concept regardless of if they are the final approach.

@neuromusic How does that sound?
